README.md

```diff
@@ -3,9 +3,10 @@ generates samplesheet compatible with TSO500 LocalApp analysis.
 
 ## Contents
 
-1. [Dependencies](#dependencies)
-2. [Description of Input Parameters](#description-of-input-parameters)
-3. [Usage](#usage)
+1. [Introduction](#introduction)
+2. [Dependencies](#dependencies)
+3. [Description of Input Parameters](#description-of-input-parameters)
+4. [Usage](#usage)
 
 ## Introduction
 
```

Comment:
Yes, agree. I think that was the original idea back then, but probably the two samplesheets diverged a bit over the time. So I am all for it that the nxf samplesheet contains all the columns needed for the samplesheet_generator and is feeded by the pipeline to it. 

> line 47: I have an idea. Maybe we can just repurpose the [nextflow samplesheet](https://github.com/InPreD/tso500_nxf_workflow/blob/main/assets/example.samplesheet.csv) which is in a tsv format (you are using pandas so it should not cause any problems to read it). Most of the information you need are in that file anyways and then we can maybe include the remaining ones. That would allow us to simple specify everything necessary in the nextflow samplesheet and in the workflow nextflow will give the samplesheet_generator file directly without any reformatting. I think that could save us some intermediate data massaging. Let me know what you think ☺️

Reply:
From my point of view, for the `sample_generator` and its usage in the `nextflow pipeline`, the most important piece of information is, that LocalApp needs a specific `illumina samplesheet` which may or may not be packed together with a sequencing run data when the data is transferred from a sequencing instrument to a compute where the `nxf pipeline` will be used to analyse the data. We should agree on the `nxf pipeline` behaviour when the `illumina samplesheet` is present (as of now, I think the best way forward would be to use the attached `illumina samplesheet` but I am not 100% sure). However, when the `illumina samplesheet` is not transferred together with the sequencing run data, `nxf pipeline` should generate the `illumina samplesheet` from the samplesheet provided as the input file to `sample_generator`. And my impression is that this main point is more or less covered in the `Intro` but feel free to point out if you think that something more should be added and what it should be, so that I better understand what you are lacking there.

Now (as far as I understand), the `illumina samplesheets` are usually filled in manually using excel and copying sample info from some database where all the info about the sequenced samples are stored. This usually happens in the unit where they run the sequencing so you can check in your node how exactly it is organised there. Also, @danielvo surely has an insight into how this is done in their node. 

It sometime makes sense to have the `illumina samplesheet` present on the sequencing machine because then you can run LocalApp analysis directly on the sequencing instrument. Additionally, when you have the `illumina samplesheet` there, you can bind it to the sequenced data and transfer it wherever you transfer the sequencing run data. 

However, in our place, there is no benefit of getting the `illumina samplesheet` on the sequencing machine, because that machine is on some sort of restricted network and the server on which we store the info about the sequenced samples is in another sort of restricted network so filling in the `illumina samplesheet` manually and transferring it then to the sequencing machine would be more pain (and no benefits) than generating the `illumina samplesheet` from scratch via a script like the `samplesheet generator`. 

Maybe over the time other nodes would see a benefit in generating the `illumina samplesheet` automatically, but maybe it won't make any sense to them for some reasons I do not see. So I do not think it makes to much sense to try to document any of this in the intro yet, but again, if you have some more concrete suggestions, just bring them in.

> When I read the introduction I am wondering why it is more efficient to generate a samplesheet from scratch. Could you include more details about the advantages and why this cli is relevant?

Reply:
sounds good, thank you ☺️

Reply:
Intro text updated in commit 3f1b418bb288edd87a39deb86bfb538e913a06f1

Reply:
Same goes for small edits in the parameter describing table. The script help got also synced with the readme description.